doc/stormond/stormond-hld.md

```diff
@@ -0,0 +1,263 @@
+- **Garbage collection:** When files are deleted or modified, the old data needs to be erased and marked as available for new data. Reserved blocks can help facilitate this process by providing a temporary location to move valid data from blocks that need to be erased. 
+
+**Temperature** - Extreme temperatures can affect SSD performance. Excessive heat can lead to throttling to prevent damage, while extreme cold can slow down data access.
+
+
+### **2.2 Priority 1 Attributes**
+
+**These are a combination of static (S) and dynamic (D) fields, offering secondary information that provides additional context about the SSD**
+
+    - Vendor Model (S)
+    - Serial Number (S)
+    - Firmware (S)
+    - Health (D)
+
+These fields are self-explanatory.
+
+
+### **2.3 `stormond` Daemon Flow**
+
+1. `stormond` would be started by the `pmon` docker container
+2. The daemon would gather the static info once init-ed, by leveraging the `SsdUtil` class and update the StateDB
+3. It would parse the priority 0 attributes by leveraging `SsdUtil` class and update the StateDB every hour.
+
+This is detailed in the sequence diagram below:
+
+![image.png](images/stormond_SequenceDiagram.png)
+
+### **2.4 Data Collection Logic**
+
+The SONiC OS already contains logic to parse information about SSDs from several vendors by way of the `SsdUtil` class. We leverage this to gather the following information:
+
+- Priority 0: Temperature
+- Priority 1: All aforementioned attributes
+
+This section will therefore only go into detail about data collection of attributes mentioned in [section 2.1](#21-priority-0-attributes).
+
+#### **2.4.1 SsdBase API additions**
+
+In order to collect IO reads/writes and number of reserved blocks, we would need to add the following member methods to the `SsdBase` class in [ssd_base.py](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_base.py) and provide a generic implementation in [ssd_generic.py](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_generic.py):
+
+
+```
+class SsdBase(object):
+
+...
+
+def get_io_reads(self):
+"""
+Retrieves the total number of Input/Output (I/O) reads done on an SSD
+
+Returns:
+    An integer value of the total number of I/O reads
+"""
+
+def get_io_writes(self):
+"""
+Retrieves the total number of Input/Output (I/O) writes done on an SSD
+
+Returns:
+    An integer value of the total number of I/O writes
+"""
+
+def get_reserved_blocks(self):
+"""
+Retrieves the total number of reserved blocks in an SSD
+
+Returns:
+    An integer value of the total number of reserved blocks
+"""
+
+```
+
+#### **2.4.2 Support for Multiple Storage Disks**
+
+The `ssdutil` utility assumes that the disk drive is `/dev/sda` whereas the drive letter could be any label based on the number of SSDs. It could also be a different type of storage device such as eMMC, USB or NVMe.
```

Comment:
@assrinivasan I think we cannot assume /dev/sda for emmC, USB

Reply:
We do not assume '/dev/sda' -- there is differentiation logic to figure out whether the block device is of type 'ata', 'usb' or 'mmcblk', and what the exact name is. Device of type 'ata' is currently supported and would instantiate an object of type 'SsdUtil'. Support for 'mmcblk' (object type 'EmmcUtil') and 'usb' (UsbUtil) are planned for future iterations. 

I will make this clearer in the document.

Reply:
Done in latest commit